ui/apps/platform/src/Components/PatternFly/BacklogListSelector.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { Key, ReactNode } from 'react';
+import {
+    Badge,
+    Button,
+    EmptyState,
+    EmptyStateIcon,
+    EmptyStateVariant,
+    Flex,
+    FormGroup,
+    Title,
+} from '@patternfly/react-core';
+import { CubesIcon, MinusCircleIcon, PlusCircleIcon } from '@patternfly/react-icons';
+
+import {
+    BaseCellProps,
+    TableComposable,
+    TableVariant,
+    Tbody,
+    Td,
+    Tr,
+} from '@patternfly/react-table';
+
+type BacklogTableProps<Item> = {
+    type: 'selected' | 'deselected';
+    label: string | undefined;
+    items: Item[];
+    listAction: (item: Item) => void;
+    rowKey: (item: Item) => Key;
+    cells: {
+        name: string;
+        render: (item: Item) => ReactNode;
+        width?: BaseCellProps['width'];
+    }[];
+    buttonText: string;
+};
+
+function BacklogTable<Item>({
+    type,
+    label,
+    items,
+    listAction,
+    rowKey,
+    cells,
+    buttonText,
+}: BacklogTableProps<Item>) {
+    const actionIcon =
+        type === 'selected' ? (
+            <MinusCircleIcon color="var(--pf-global--danger-color--200)" />
+        ) : (
+            <PlusCircleIcon color="var(--pf-global--primary-color--100)" />
+        );
+    return (
+        <FormGroup
+            label={
+                <>
+                    {label}
+                    <Badge className="pf-u-ml-sm" isRead>
+                        {items.length}
+                    </Badge>
```

Comment:
Thoughts about badge, which I was happy to see after designs had moved away from it for a while:
* Attached collections: Does search filter apply to this table? Length seems meaningful only if not filtered.
* Detached collections: Length seems meaningful only if `!hasMore` and maybe only without search filter.

Reply:
Agreed that ht badge has limited utility when the full dataset is not present, and the one case where it would be meaningful (unfiltered attached collections) is likely going to be a small enough number that the badge is fairly unimportant.

I left the ability to show the badge in the underlying component though, since if we get reuse out of it there is the possibility we use it against data that is complete and unfilterable.